nemo/core/classes/modelPT.py

```diff
@@ -737,6 +749,15 @@ def prepare_test(self, trainer: 'Trainer') -> bool:
 
         return True
 
+    def set_trainer(self, trainer: 'Trainer'):
+        """
+        Set an instance of Trainer object.
+
+        Args:
+            trainer: PyTorch Lightning Trainer object.
+        """
+        self._trainer = trainer
```

Comment:
Is this new function used anywhere?

Reply:
Right the new function is not used yet, but should be used after restoration and before calling trainer.fit(). This will allow schedulers to properly be built (since they depend on trainer object). I think once schedulers are built, the trainer restore_from_checkpoint will initialize the optimizers and schedulers but I havent tested this behaviour yet. If it turns out setting up trainer in the restored model is unnecessary, we can remove this safely.